theories/lra.elpi

```diff
@@ -100,8 +100,8 @@ rfstr.prop {{ @eq lp:Bool lp:Ty1 lp:Ty2 }} C IsField :-
 rfstr.prop {{ @eq lp:Ty _ _ }} C IsField :- !, type->carrier Ty C IsField.
 
 pred rfstr.hyps i:list prop, o:carrier, o:prop.
-rfstr.hyps [decl _ _ H|_] C IsField :- rfstr.prop H C IsField.
 rfstr.hyps [_|Ctx] C IsField :- rfstr.hyps Ctx C IsField.
+rfstr.hyps [decl _ _ H|_] C IsField :- rfstr.prop H C IsField.
```

Comment:
why don't you rev the list instead?

Reply:
I rather removed `std.rev` (in L.456). Do you think it is better to use rev?

Reply:
Since I did not see why we need to take the reversal of `Ctx` for `quote.goal`, I wanted to remove it entirely.

Reply:
I never tested the perf of exists+rev v.s. the code you wrote which is more heavy on backtracking....
I just find it surprising

Reply:
Sure. I will fix.